circuit-types/src/keychain.rs

```diff
@@ -129,105 +140,63 @@ impl From<SecretIdentificationKey> for Scalar {
     linkable,
     secret_share
 )]
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub struct NonNativeKey<const KEY_WORDS: usize> {
-    /// The `Scalar` words used to represent the key
-    ///
-    /// Because the key is a point on a non-native curve, its representation requires
-    /// a bigint like approach
-    pub key_words: [Scalar; KEY_WORDS],
-}
-
-impl<const KEY_WORDS: usize> Default for NonNativeKey<KEY_WORDS> {
-    fn default() -> Self {
-        Self {
-            key_words: [Scalar::zero(); KEY_WORDS],
-        }
-    }
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PublicSigningKey {
+    pub x: [Scalar; FELT_WORDS],
+    pub y: [Scalar; FELT_WORDS],
 }
 
-impl<const KEY_WORDS: usize> NonNativeKey<KEY_WORDS> {
-    /// Split a biguint into scalar words in little endian order
-    fn split_biguint_into_words(mut val: BigUint) -> [Scalar; KEY_WORDS] {
-        let scalar_mod = get_scalar_field_modulus();
-        let mut res = Vec::with_capacity(KEY_WORDS);
-        for _ in 0..KEY_WORDS {
-            let word = Scalar::from(&val % &scalar_mod);
-            val /= &scalar_mod;
-            res.push(word);
-        }
+impl PublicSigningKey {
+    /// Split a biguint into 128-bit scalar words
+    fn split_biguint_into_words(val: &BigUint) -> [Scalar; FELT_WORDS] {
+        let low_mask = val & &*LOW_BIT_MASK;
+        let low: u128 = low_mask.try_into().unwrap();
+        let high: u128 = (val >> 128u32).try_into().unwrap();
 
-        res.try_into().unwrap()
+        [Scalar::from(low), Scalar::from(high)]
     }
```

Comment:
A couple of suggestions on improving this method:
1. It's much cleaner to define a modulus and use `div_rem`
2. We don't need to go through `u128` to get to `Scalar`
```rust
use renegade_crypto::fields::biguint_to_scalar;

lazy_static {
     static ref TWO_TO_128: BigUint = BigUint::from(1u8) << 128;
}

fn split_biguint_into_words(val: &BigUint) -> [Scalar; FELT_WORDS] {
    let (high, low) = BigUint::div_rem(val, &TWO_TO_128);

    [
        biguint_to_scalar(&low),
        biguint_to_scalar(&high),
    ]
}
```